Restart SSE connection in listenForItemChange() when not alive

mueller-ma opened the pull request:
When restarting openHAB existing SSE connections may look like they are still intact to the client, but no further events are received. Thus restart a connection when no ALIVE events are received anymore.

mueller-ma committed:
Restart SSE connection in listenForItemChange() when not alive

When restarting openHAB existing SSE connections may look like they are still intact to the client, but no further events are received. Thus restart a connection when no ALIVE events are received anymore.

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

maniac103 commented on mobile/src/main/java/org/openhab/habdroid/util/ItemClient.kt:
```diff
@@ -119,10 +121,33 @@ object ItemClient {
 
+        suspend fun restartSubscription() {
+            try {
+                eventSubscription.cancel()
```
'Raised' as in 'without the try-catch'? How exactly does the exception (message) look like?

mueller-ma replied:
You can test this by setting the delay in line 139 to less than 10 seconds.

maniac103 replied:
@mueller-ma Any thoughts on the patch? In general I think the issue is worth fixing in this beta cycle. FWIW, I rebased my patch to apply cleanly, it's the top one in [this branch](https://github.com/maniac103/openhab.android/tree/restart-sse).

mueller-ma replied:
Your patch looks good 👍 

I picked it into my branch, so it should be fine to merge this.

maniac103 committed:
Refactor ItemClient.listenForItemChange

Make it use a channel instead of callbacks, and avoid
ClosedReceiveChannelException being thrown when restarting the SSE
connection.

Signed-off-by: Danny Baumann <dannybaumann@web.de>
